Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Endpoints/AWSEndpointConfig.swift

```diff
@@ -0,0 +1,26 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+@_spi(FileBasedConfig) import AWSSDKCommon
+
+public enum AWSEndpointConfig {
+
```

Comment:
Function below attempts to resolve account ID endpoint mode from env var & config file, and if no luck defaults to `.preferred` mode.